docs/under-the-hood/storage.md

```diff
@@ -1,138 +1,201 @@
-changed at any time. 
+Depending on the actual value stored, each `LabelPropertyIndex::Entry` has at least **72B**.
 
-Relationships have a type, also represented as text. Unlike nodes, relationships
-must have exactly one relationship type and once it is set upon creation, it can
-never be modified again.
+Each Index Entry object is placed into the Skiplist.
+
+#### Each index object in total
+
+- `SkipListNode<LabelIndex::Entry>` object has **24B**.
+- `SkipListNode<LabelPropertyIndex::Entry>` has at least **80B**.
+- Each `SkipListNode` has an additional **16B** because of the **next_pointers**.
 
 ### Properties
 
-Nodes and relationships can store various properties. Properties are similar to
-mappings or tables containing property names and their accompanying values.
-Property names are represented as text, while values can be of different types.
-Each property can store a single value, and it is not possible to have multiple
-properties with the same name on a single graph element. But, the same property
-names can be found across multiple graph elements. Also, there are no
-restrictions on the number of properties that can be stored in a single graph
-element. The only restriction is that the values must be of the supported types.
-Below is a table of supported data types.
-
- Type      | Description
------------|------------
- `Null`    | Property has no value, which is the same as if the property does not exist.
- `String`  | A character string (text).
- `Boolean` | A boolean value, either `true` or `false`.
- `Integer` | An integer number.
- `Float`   | A floating-point number (real number).
- `List`    | A list containing any number of property values of any supported type under a single property name.
- `Map`     | A mapping of string keys to values of any supported type.
- `Duration`| A period of time.
- `Date`    | A date with year, month, and day.
- `LocalTime` | Time within a day without timezone.
- `LocalDateTime` | Date and local time.
-
-Check the reference guide to [read more about temporal
-types](/reference-guide/data-types.md) `Duration`, `Date`, `LocalTime` and
-`LocalDateTime`. 
-
-:::note
-
-Even though it's possible to store `List` and `Map` property values, it is
-impossible to modify them. It is, however, possible to replace them entirely.
-So, the following queries are valid:
-
-```cypher
-CREATE (:Node {property: [1, 2, 3]});
-CREATE (:Node {property: {key: "value"}});
-```
-
-But these are not:
-
-```cypher
-MATCH (n:Node) SET n.property[0] = 0;
-MATCH (n:Node) SET n.property.key = "other value";
-```
+All properties use at least **1B** for encoding the extra information about the
+type of the value stored, the size of the property key (is an integer stored in
+1B, 2B, 4B, or 8B), and the last **4bits** are used for a variety of reasons
+depending on the value type of the property. After the initial byte, properties
+store the value of its key.
```

Comment:
I am confused with this part, this `property key` is not clear to me. @vpavicic we could as Marko (when asking for other stuff) to clarify this a bit more, so maybe we could write it more clearly.

Reply:
i'm not sure what are you unclear about, which statement?

Reply:
It is unclear how it can be at least 1B.  If you have 1B  integer property key + 4 bits for something. Than it is always at least 2B.  I do not understand this property key structure, that is why it is unclear. But this is great start, we can be more precise when distributed kicks in.

Reply:
i understood that encoding the extra info takes up at least 1B?

Reply:
Yep, leave as it is, but property key should be explained more in detail.